src/en/manhwa18/src/eu/kanade/tachiyomi/extension/en/manhwa18/Manhwa18.kt

```diff
@@ -1,62 +1,217 @@
+                            addQueryParameter("type", filter.checked)
+                        }
+                    }
+                    is NationFilter -> {
+                        if (filter.checked.isNotBlank()) {
+                            addQueryParameter("nation", filter.checked)
+                        }
+                    }
+                    is SortFilter -> {
+                        addQueryParameter("arrange", filter.getValue())
+                    }
+                    is StatusFilter -> {
+                        addQueryParameter("is_complete", filter.getValue())
+                    }
+                    else -> {}
+                }
+            }
+        }
+        return GET(url.build(), headers)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        return GET("$apiUrl/get-search-suggest/$query", headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        throw UnsupportedOperationException()
+    }
+
+    private fun searchMangaParsePagination(page: Int, response: Response): MangasPage {
+        val fromIndex = (page - 1) * MAX_MANGA_PER_PAGE
+        val toIndex = page * MAX_MANGA_PER_PAGE
+        val result = json.decodeFromString<List<Manga>>(response.body.string())
+        return MangasPage(
+            result
+                .subList(
+                    fromIndex,
+                    min(toIndex, result.size),
+                )
+                .map { manga ->
+                    manga.toSManga()
+                },
+            hasNextPage = toIndex < result.size,
+        )
+    }
+
+    // manga details
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        val slug = manga.url.substringAfterLast('/')
+        return GET("$apiUrl/get-detail-product/$slug", headers)
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val mangaDetail = json.decodeFromString<MangaDetail>(response.body.string())
+        return mangaDetail.manga.toSManga().apply {
+            initialized = true
+        }
+    }
+
+    override fun getMangaUrl(manga: SManga): String {
+        return "${baseUrl}${manga.url}"
+    }
+
+    // chapter list
+    override fun chapterListRequest(manga: SManga): Request {
+        return mangaDetailsRequest(manga)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val mangaDetail = json.decodeFromString<MangaDetail>(response.body.string())
+        val mangaSlug = mangaDetail.manga.slug
+
+        return mangaDetail.manga.episodes?.map { chapter ->
+            SChapter.create().apply {
+                // compatible with old theme
+                setUrlWithoutDomain("/manga/$mangaSlug/${chapter.slug}")
+                name = chapter.name
+                date_upload = chapter.created_at?.parseDate() ?: 0L
+                chapter_number = chapter.name.toFloat()
```

Comment:
app will try to set the number itself, check if that one works correctly for this source